swarm/storage/chunker.go

```diff
@@ -262,8 +253,11 @@ func (self *TreeChunker) hashWorker(jobC chan *hashJob, chunkC chan *Chunk, errC
 // - the size (of the subtree encoded in the Chunk)
 // - the Chunk, ie. the contents read from the input reader
 func (self *TreeChunker) hashChunk(hasher hash.Hash, job *hashJob, chunkC chan *Chunk, swg *sync.WaitGroup) {
+
 	hasher.Write(job.chunk)
 	h := hasher.Sum(nil)
+	hasher.Reset()
```

Comment:
Reset should come before and the bmt should be fixed to
reserve tree from pool with `Reset()` and release on `Sum()`